source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/V1/ChargePricesControllerTests.cs

```diff
@@ -0,0 +1,122 @@
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Energinet.DataHub.Charges.Contracts.ChargePrice;
+using FluentAssertions;
+using FluentAssertions.Common;
+using GreenEnergyHub.Charges.IntegrationTest.Core.Fixtures.Database;
+using GreenEnergyHub.Charges.IntegrationTest.Core.Fixtures.WebApi;
+using GreenEnergyHub.Charges.TestCore.Attributes;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Categories;
+using JsonSerializerOptions = System.Text.Json.JsonSerializerOptions;
+
+namespace GreenEnergyHub.Charges.IntegrationTests.IntegrationTests.WebApi.V1
+{
+    [IntegrationTest]
+    [Collection(nameof(ChargesWebApiCollectionFixture))]
+    public class ChargePricesControllerTests :
+        WebApiTestBase<ChargesWebApiFixture>,
+        IClassFixture<ChargesWebApiFixture>,
+        IClassFixture<WebApiFactory>
+    {
+        private const string BaseUrl = "/v1/ChargePrices";
+        private readonly ChargesDatabaseManager _databaseManager;
+
+        public ChargePricesControllerTests(
+            ChargesWebApiFixture chargesWebApiFixture,
+            ITestOutputHelper testOutputHelper)
+            : base(chargesWebApiFixture, testOutputHelper)
+        {
+            _databaseManager = chargesWebApiFixture.DatabaseManager;
+        }
+
+        [Theory]
+        [InlineAutoMoqData]
+        public async Task SearchAsync_WhenChargePricesExists_ReturnsOkAndCorrectContentType(
+            WebApiFactory factory)
+        {
+            // Arrange
+            var sut = CreateHttpClient(factory);
+            var searchCriteria = new ChargePricesSearchCriteriaV1Dto(Guid.NewGuid(), DateTime.Now, DateTime.Now);
+
+            // Act
+            var response = await sut.PostAsJsonAsync($"{BaseUrl}/Search", searchCriteria);
+
+            // Assert
+            var contentType = response.Content.Headers.ContentType!.ToString();
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            contentType.Should().Be("application/json; charset=utf-8");
+        }
+
+        [Theory]
+        [InlineAutoMoqData]
+        public async Task SearchAsync_WhenRequested_ReturnsChargePrices(WebApiFactory factory)
+        {
+            // Arrange
+            await using var chargesDatabaseContext = _databaseManager.CreateDbQueryContext();
+
+            var charge = chargesDatabaseContext.Charges.Include(c => c.ChargePoints).Single(c => c.SenderProvidedChargeId == "TestTariff");
+            var chargePoint = charge.ChargePoints.First();
+
+            var sut = CreateHttpClient(factory);
+            var searchCriteria =
+                new ChargePricesSearchCriteriaV1Dto(
+                    chargePoint.ChargeId,
+                    chargePoint.Time.AddDays(-1),
```

Comment:
Let's keep it at the boundary value

```suggestion
                    chargePoint.Time,
```